src/promo/promo.service.ts

```diff
@@ -52,7 +52,7 @@ export class PromocodeService {
     try {
       const promocode: IPromoCode = await this.promocodeRepository.getPromoCodeByName(promocodeName);
       if (!promocode) return false;
-      if (promocode.deletedAt !== null) return false;
+      if (promocode.deletedAt !== null && promocode.deletedAt !== undefined) return false;
```

Comment:
why not to remove for false check? I mean   if(promocode.deletedAt) return false;